tests/testthat/test-3agglomerate.R

```diff
@@ -29,123 +29,148 @@ test_that("agglomerate", {
+    altExp(tse, "Family2") <- agglomerateByRank(
+        tse, rank="Family", na.rm = FALSE)
+    altExp(tse, "Family3") <- agglomerateByRank(
+        tse, rank="Family", onRankOnly = TRUE, na.rm = TRUE)
+    altExp(tse, "Family4") <- agglomerateByRank(
+        tse, rank="Family", onRankOnly = TRUE, na.rm = FALSE)
+    altExp(tse, "Family5") <- agglomerateByVariable(
+        tse, f="Family", MARGIN = "row", na,rm = FALSE)
     
-    # Other group is added by agglomerateByPrevalence function to collect features under threshold
-    actual <- agglomerateByPrevalence(tse, rank="Family", assay.type="relabundance", 
-                                         detection  = 0.5/100, prevalence  = 20/100)
-    actual0 <- agglomerateByPrevalence(altExp(tse, "Family"), assay.type="relabundance", 
-                                       detection  = 0.5/100, prevalence  = 20/100)
-    actual1 <- agglomerateByPrevalence(altExp(tse, "Family1"), assay.type="relabundance", 
-                                       detection  = 0.5/100, prevalence  = 20/100)
-    actual2 <- agglomerateByPrevalence(altExp(tse, "Family2"), assay.type="relabundance", 
-                                       detection  = 0.5/100, prevalence  = 20/100)
-    actual3 <- agglomerateByPrevalence(altExp(tse, "Family3"), assay.type="relabundance", 
-                                       detection  = 0.5/100, prevalence  = 20/100)
-    actual4 <- agglomerateByPrevalence(altExp(tse, "Family4"), assay.type="relabundance", 
-                                       detection  = 0.5/100, prevalence  = 20/100)
-    actual5 <- agglomerateByPrevalence(altExp(tse, "Family5"), assay.type="relabundance", 
-                                       detection  = 0.5/100, prevalence  = 20/100)
+    # Other group is added by agglomerateByPrevalence function to collect
+    # features under threshold
+    actual <- agglomerateByPrevalence(
+        tse, rank = "Family", assay.type = "relabundance", detection  = 0.5/100,
+        prevalence  = 20/100, agg.na.rm = TRUE)
+    actual0 <- agglomerateByPrevalence(
+        altExp(tse, "Family"), assay.type="relabundance",
+        detection  = 0.5/100, prevalence  = 20/100)
+    actual1 <- agglomerateByPrevalence(
+        altExp(tse, "Family1"), assay.type="relabundance", detection  = 0.5/100,
+        prevalence  = 20/100)
+    actual2 <- agglomerateByPrevalence(
+        altExp(tse, "Family2"), assay.type="relabundance", detection  = 0.5/100,
+        prevalence  = 20/100)
+    actual3 <- agglomerateByPrevalence(
+        altExp(tse, "Family3"), assay.type="relabundance", detection  = 0.5/100,
+        prevalence  = 20/100)
+    actual4 <- agglomerateByPrevalence(
+        altExp(tse, "Family4"), assay.type="relabundance", detection  = 0.5/100,
+        prevalence  = 20/100)
+    actual5 <- agglomerateByPrevalence(
+        altExp(tse, "Family5"), assay.type="relabundance", detection  = 0.5/100,
+        prevalence  = 20/100)
```

Comment:
Can you simplify these tests? It is very hard for developer to say what they are actually testing.

For example, agglomerateByPrevalence is just additional layer of complexity. Tests should:

1. Test that correct values are calculated
2. Be easy to read, so that we can easily debug what is the problem

The tests should be like this. This is easy to read what is the purpose:

```
test <- agglomerateByRank(tse, rank="Family", na.rm = TRUE)
ref <- unique(test$Family)
ref <- ref[ !is.na(ref) ]
ref <- length(ref)
expect_equal(nrow(test), ref)
```

Reply:
ACK.